src/ScriptCs.Core/ScriptLibraryComposer.cs

```diff
@@ -0,0 +1,129 @@
+
+            _fileSystem = fileSystem;
+            _preProcessor = preProcessor;
+            _packageContainer = packageContainer;
+            _packageAssemblyResolver = packageAssemblyResolver;
+            _logger = logger;
+        }
+
+        internal string GetMainScript(IPackageObject package)
+        {
+            var content = package.GetContentFiles()
+                .Where(f => f.EndsWith("main.csx", StringComparison.InvariantCultureIgnoreCase)).ToArray();
+
+            string script = null;
+            var count = content.Length;
+
+            if (count == 1)
+            {
+                script = content[0];
+            } 
+            else if (content.Count() > 1)
+            {
+                _logger.WarnFormat("Script Libraries in '{0}' ignored due to multiple Main files being present", package.FullName);
+                return null;
+            }
+           
+            if (script != null)
+            {
+                _logger.DebugFormat("Found main script: {0}", script);    
+            }
+            
+            return script;
+        }
+
+        public virtual string ScriptLibrariesFile
+        {
+            get { return "ScriptLibraries.csx"; }
+        }
+
+        public void Compose(string workingDirectory, StringBuilder builder = null)
+        {
+            var namespaces = new List<string>();
+            var references = new List<string>();
+
+            var packagesPath = Path.Combine(workingDirectory, _fileSystem.PackagesFolder);
+            var packageReferences = _packageAssemblyResolver.GetPackages(workingDirectory);
+            var packageScriptsPath = Path.Combine(packagesPath, ScriptLibrariesFile);
+
+            if (!_fileSystem.DirectoryExists(packagesPath) || _fileSystem.FileExists(packageScriptsPath))
+            {
+                return;
+            }
+
+            if (builder == null)
+            {
+                builder = new StringBuilder();
+            }
+
+            foreach (var reference in packageReferences)
+            {
+                ProcessPackage(packagesPath, reference, builder, references, namespaces);
+            }
+
+            foreach (var ns in namespaces)
+            {
+                builder.Insert(0, String.Format("using {0};{1}", ns, Environment.NewLine));
+            }
+
+            foreach (var reference in references)
+            {
+                builder.Insert(0, String.Format("#r {0}{1}", reference, Environment.NewLine));
+            }
+            _fileSystem.WriteToFile(packageScriptsPath, builder.ToString());
+        }
+
+        protected internal virtual void ProcessPackage(string packagesPath, IPackageReference reference, StringBuilder builder, List<string> references,
+            List<string> namespaces)
+        {
+            _logger.DebugFormat("Finding package:{0}", reference.PackageId);
+            var package = _packageContainer.FindPackage(packagesPath, reference);
```

Comment:
If a package is missing from the packages directory, a null reference is returned here. It is then passed `GetMainScript()`, which throws an NRE, which kills the app.